pkg/kubelet/prober/prober_manager.go

```diff
@@ -264,39 +275,81 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 			started = result == results.Success
 		} else {
 			// The check whether there is a probe which hasn't run yet.
-			_, exists := m.getWorker(podUID, c.Name, startup)
+			_, exists := m.getWorker(pod.UID, c.Name, startup)
 			started = !exists
 		}
 		podStatus.ContainerStatuses[i].Started = &started
 
-		if started {
-			var ready bool
-			if c.State.Running == nil {
-				ready = false
-			} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok && result == results.Success {
-				ready = true
-			} else {
-				// The check whether there is a probe which hasn't run yet.
-				w, exists := m.getWorker(podUID, c.Name, readiness)
-				ready = !exists // no readinessProbe -> always ready
-				if exists {
-					// Trigger an immediate run of the readinessProbe to update ready state
-					select {
-					case w.manualTriggerCh <- struct{}{}:
-					default: // Non-blocking.
-						klog.InfoS("Failed to trigger a manual run", "probe", w.probeType.String())
-					}
+		if !started {
+			continue
+		}
+
+		var ready bool
+		if c.State.Running == nil {
+			ready = false
+		} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok && result == results.Success {
+			ready = true
+		} else {
+			// The check whether there is a probe which hasn't run yet.
+			w, exists := m.getWorker(pod.UID, c.Name, readiness)
+			ready = !exists // no readinessProbe -> always ready
+			if exists {
+				// Trigger an immediate run of the readinessProbe to update ready state
+				select {
+				case w.manualTriggerCh <- struct{}{}:
+				default: // Non-blocking.
+					klog.InfoS("Failed to trigger a manual run", "probe", w.probeType.String())
 				}
 			}
-			podStatus.ContainerStatuses[i].Ready = ready
 		}
+		podStatus.ContainerStatuses[i].Ready = ready
+	}
+
+	initContainers := make(map[string]*v1.Container, len(pod.Spec.InitContainers))
+	for i := range pod.Spec.InitContainers {
+		initContainers[pod.Spec.InitContainers[i].Name] = &pod.Spec.InitContainers[i]
 	}
-	// init containers are ready if they have exited with success or if a readiness probe has
-	// succeeded.
 	for i, c := range podStatus.InitContainerStatuses {
+		var started bool
```

Comment:
This is now duplicated as the same exact lines as between 270-281 (for regular non-init containers). Can we extract this out to another function and re-use it to avoid the duplication?

Reply:
added the `isContainerStarted` function